components/support/nimbus-cli/src/sources/experiment.rs

```diff
@@ -202,12 +231,36 @@ impl TryFrom<&ExperimentSource> for Value {
                 files,
             } => feature_utils::create_experiment(app, feature_id, files)?,
 
+            ExperimentSource::WithPatchFile { patch, inner } => patch_experiment(inner, patch)?,
+
             #[cfg(test)]
             ExperimentSource::FromTestFixture { file } => value_utils::read_from_file(file)?,
         })
     }
 }
 
+fn patch_experiment(experiment: &ExperimentSource, patch: &PathBuf) -> Result<Value> {
+    let mut value: Value = experiment.try_into()?;
+
+    let patch: FeatureDefaults = read_from_file(patch)?;
```

Comment:
So I initially hit an issue with this in that I wrote the patch file in yaml but didn't include the `yaml` extension on the file. 

The error just said something along the lines of "ident error line 1 column 2". It'd be nice if we could add a map_err here and say that there's an error with the patch file so the user knows where the problem lies.